test/runtime_tests.cc

```diff
@@ -1064,5 +1064,40 @@ namespace detail {
 		});
 	}
 
+	TEST_CASE_METHOD(test_utils::runtime_fixture, "0-dimensional kernels work as expected", "[buffer]") {
+		constexpr float value_a = 13.37f;
+		constexpr float value_b = 42.0f;
+
+		buffer<float, 0> buf_a(value_a);
```

Comment:
We should probably also have a test (or do this here?) of a 0-dimensional accessing a non-0-dimensional buffer!

Reply:
I've added a test, and also moved accessor-related tests to `accessor_tests.cc`.